### ISSUE TITLE
restrict-plus-operands doesn't recognize property as string type

### ISSUE THREAD

**Issue by NN---** [palantir/tslint#3012]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.4.1
- __Running TSLint via__: gulp

#### TypeScript code being linted

```ts
var obj = { ... };
for (var p in obj) {  // p is typed as string in TypeScript
  var str: string = p + "Anything";
}
```

with `tslint.json` configuration:

```json
restrict-plus-operands: true
```

#### Actual behavior
error restrictPlusOperands: Operands of '+' operation must either be both strings or both numbers

#### Expected behavior
No error

**Comment by KSuttle-cng** [comment]
It also doesn't ignore the TS parseInt operator 

```ts
let numStr = '32'; 
let numStrWithPlus = +numStr; // <== Error: restrict-plus-operands
```

**Comment by JoshuaKGoldberg** [comment]
@NN--- I'm not able to reproduce this with TSLint 5.11.0 and the code you posted. Perhaps the bad behavior you were seeing was from a version mismatch? Resolving as aged away, but please do post if you're able to demonstrate the bug still happening!
